### PR TITLE
feat(fastly_secretstore): implement resource and documentation

### PR DESCRIPTION
**Pull request by Integralist**
This PR implements a new `fastly_secretstore` resource and `fastly_secretstores` data source.

✅ The full end-to-end integration test suite has been run and is passing successfully.

### CODE REVIEW

File: fastly/resource_fastly_secretstore_test.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+			ServiceVersion: service.Version.Number,
+		})
+		if err != nil {
+			return fmt.Errorf("error listing all resource links: %s", err)
+		}
+
+		found = false
+		for _, link := range links {
+			if link.Name == linkName {
+				found = true
```
Comment (mccurdyc) [inline fastly/resource_fastly_secretstore_test.go:101]: Why don't you `break` on this one?

**Review by mccurdyc — APPROVED** [review]
One non-blocking question

File: fastly/resource_fastly_secretstore_test.go (hunk 1 above)
Comment (Integralist) [reply fastly/resource_fastly_secretstore_test.go:101]: Good spot, I should break there yes!

**Review by mccurdyc — APPROVED** [review]

File: fastly/resource_fastly_secretstore.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	log.Printf("[DEBUG] REFRESH: Secret Store input: %#v", input)
+
+	store, err := conn.GetSecretStore(input)
+	if err != nil {
+		log.Printf("[WARN] No Secret Store found '%s'", d.Id())
+		d.SetId("")
+		return diag.FromErr(err)
+	}
```
Comment (Integralist) [inline fastly/resource_fastly_secretstore.go:65]:
```suggestion
	if err != nil {
		if e, ok := err.(*gofastly.HTTPError); ok && e.IsNotFound() {
			log.Printf("[WARN] No Secret Store found '%s'", d.Id())
			d.SetId("")
			return nil
		}
		return diag.FromErr(err)
	}
```

**Review by mccurdyc — APPROVED** [review]

**Comment by jfctfl** [comment]
Hi, glad to see this coming! We could really use this for our terraform deployment, are there any plans on when this will be merged and available?

**Comment by Integralist** [comment]
Thanks @jfctfl 

I'm just at this moment chasing up feedback on when I'm able to merge this PR and publish it into a new release.

Unfortunately I don't have a timeline yet. Hopefully not too long.

**Comment by jfctfl** [comment]
Awesome 🚀 
If you have any updates (also on the timeline), would appreciate if you could share those!

**Pull request merged**